app/router.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+import * as steps from './steps.js';
+
+/* components */
+import GoalsList from './components/goals-list/goals-list.jsx';
+import AddGoal from './components/add-new-goal/add-new-goal.jsx';
+import ViewGoal from './components/view-goal/view-goal.jsx';
+
+export default (props) => {
+
+  switch (props.step) {
+    case steps.ADD_GOAL:
+      const newGoalId = (props.goals.map(g => g.id).sort((a, b) => b - a)[0] + 1 || 0);
+      return <AddGoal
+        newGoal = { props.newGoal }
+        onInputGoal={ props.onInputGoal }
+        onSelectAvatar={ props.onSelectAvatar }
+        saveNewGoal= { props.saveNewGoal }
+        newGoalId= { newGoalId }
```

Comment:
I am confused here. We already have something called `newGoal`, but we are also doing a search for the newGoal id on the goals. I think this is either bad naming or a mistake